os/drivers/audio/alc5658.c

```diff
@@ -950,15 +949,19 @@ static int alc5658_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned lo
 
 		/* Report our preferred buffer size and quantity */
 
-#ifdef CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS
 	case AUDIOIOC_GETBUFFERINFO: {
 		audvdbg("AUDIOIOC_GETBUFFERINFO:\n");
 		bufinfo = (FAR struct ap_buffer_info_s *)arg;
+#ifdef CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS
 		bufinfo->buffer_size = CONFIG_ALC5658_BUFFER_SIZE;
 		bufinfo->nbuffers = CONFIG_ALC5658_NUM_BUFFERS;
+#else
+		bufinfo->buffer_size = priv->samprate / (1000 / CONFIG_ALC5658_I2S_TIMEOUT);
+		bufinfo->nbuffers = CONFIG_AUDIO_NUM_BUFFERS;
```

Comment:
If we set CONFIG_AUDIO_NUM_BUFFERS = 8, then we cannot use it here. Since we are calculating buffer size based on sample rate, we can set numbuffers to 2.

Reply:
That is user fault, we don't have to prevent this kind of user fault.
But user should test and find best configuration values.
we just suggested 2 as a default setting.